invoices/invoiceregistry_test.go

```diff
@@ -20,11 +21,115 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// TestSettleInvoice tests settling of an invoice and related notifications.
-func TestSettleInvoice(t *testing.T) {
+// TestInvoiceRegistry is a master test which encompasses all tests using an
+// InvoiceDB instance. The purpose of this test is to be able to run all tests
+// with a custom DB instance, so that we can test the same logic with different
+// DB implementations.
+func TestInvoiceRegistry(t *testing.T) {
```

Comment:
Great idea 👍

Reply:
I think it's great to be able to run the quick and simple to debug unit tests on all backends. This ha been essential for me when porting invoice db to SQL, otherwise i guess iteration speed would have been much slower.

I see your concern though, but I feel like given that the `InvoiceDB` interface itself is defined in the `invoices` package it's probably best to keep the tests here and implementation details elsewhere. Another option (for the future) might be to move the k/v implementation under `invoices/kvdb` and the sql implementation under `invoices/sql`.